### ISSUE TITLE
🐛 Bug Report: `RequirePermission` causes "Mounted routable extension must have a path"

### ISSUE THREAD

**Issue by dhoizner** [backstage/backstage#15365]
### 📜 Description

When attempting to wrap routes in `RequirePermission` as follows:

```
      <Route path="/route-here" element={
        <RequirePermission permission={myPermissionHere}>
          <MyComponentHere />
        </RequirePermission>
      } />
```

I am encountering the following error:

```
Error
Mounted routable extension must have a path
Call Stack
 Object.eval [as visit]
  node_modules/@backstage/core-app-api/dist/index.esm.js:1981:15
 eval
  node_modules/@backstage/core-app-api/dist/index.esm.js:1752:42
 eval
  node_modules/react/cjs/react.development.js:1104:17
 eval
  node_modules/react/cjs/react.development.js:1067:17
 mapIntoArray
  node_modules/react/cjs/react.development.js:964:23
 mapChildren
  node_modules/react/cjs/react.development.js:1066:3
 Object.forEachChildren [as forEach]
  node_modules/react/cjs/react.development.js:1103:3
 traverseElementTree
  node_modules/@backstage/core-app-api/dist/index.esm.js:1744:49
 eval
  node_modules/@backstage/core-app-api/dist/index.esm.js:2648:24
 mountMemo
  node_modules/react-dom/cjs/react-dom.development.js:15846:19
```

This error does not occur when removing the `RequirePermission` parent tag.

I would be happy to submit a PR for this, but I'm not sure where the problem is.

### 👍 Expected behavior

Successfully render the application, displaying the component when the permission check passes and displaying an error page when it doesn't.

### 👎 Actual Behavior with Screenshots

![Screenshot 2022-12-21 at 3 12 27 PM](https://user-images.githubusercontent.com/419620/208913422-14bc76a3-36bd-4449-a9f1-6eb8f3ca3442.png)

### 👟 Reproduction steps

Introducing a `RequirePermission` tag into any `Route` tag element within `packages/app/src/App.tsx` causes the above error to occur.

### 🖥️ Your Environment

Browser: Chrome Version 107.0.5304.121 (Official Build) (x86_64)

`yarn backstage-cli info`:
```
yarn run v1.22.19
$ /Users/dan.hoizner/git/collibra/portal/node_modules/.bin/backstage-cli info
OS:   Darwin 22.1.0 - darwin/x64
node: v16.15.1
yarn: 1.22.19
cli:  0.20.0 (installed)
backstage:  1.7.0

Dependencies:
  @backstage/app-defaults                          1.0.7
  @backstage/backend-common                        0.15.2, 0.16.0, 0.17.0
  @backstage/backend-plugin-api                    0.1.4
  @backstage/backend-tasks                         0.3.7
  @backstage/catalog-client                        1.1.2
  @backstage/catalog-model                         1.1.3
  @backstage/cli-common                            0.1.11
  @backstage/cli                                   0.20.0
  @backstage/config-loader                         1.1.7
  @backstage/config                                1.0.5
  @backstage/core-app-api                          1.1.1
  @backstage/core-components                       0.11.2
  @backstage/core-plugin-api                       1.2.0
  @backstage/dev-utils                             1.0.7
  @backstage/errors                                1.1.4
  @backstage/integration-react                     1.1.5
  @backstage/integration                           1.4.1
  @backstage/plugin-api-docs                       0.8.10
  @backstage/plugin-app-backend                    0.3.37
  @backstage/plugin-auth-backend                   0.17.0
  @backstage/plugin-auth-node                      0.2.8
  @backstage/plugin-catalog-backend                1.5.1
  @backstage/plugin-catalog-common                 1.0.8
  @backstage/plugin-catalog-graph                  0.2.22
  @backstage/plugin-catalog-import                 0.9.0
  @backstage/plugin-catalog-node                   1.2.1
  @backstage/plugin-catalog-react                  1.2.0
  @backstage/plugin-catalog                        1.6.0
  @backstage/plugin-github-actions                 0.5.10
  @backstage/plugin-org                            0.5.10
  @backstage/plugin-permission-backend             0.5.14
  @backstage/plugin-permission-common              0.7.2
  @backstage/plugin-permission-node                0.7.2
  @backstage/plugin-permission-react               0.4.8
  @backstage/plugin-proxy-backend                  0.2.31
  @backstage/plugin-scaffolder-backend             1.7.0
  @backstage/plugin-scaffolder-common              1.2.2
  @backstage/plugin-scaffolder                     1.7.0
  @backstage/plugin-search-backend-module-pg       0.4.1
  @backstage/plugin-search-backend-node            1.0.3
  @backstage/plugin-search-backend                 1.1.0
  @backstage/plugin-search-common                  1.1.1
  @backstage/plugin-search-react                   1.2.0
  @backstage/plugin-search                         1.0.3
  @backstage/plugin-tech-radar                     0.5.17
  @backstage/plugin-techdocs-backend               1.4.0
  @backstage/plugin-techdocs-module-addons-contrib 1.0.5
  @backstage/plugin-techdocs-node                  1.4.1
  @backstage/plugin-techdocs-react                 1.0.5
  @backstage/plugin-techdocs                       1.3.3
  @backstage/plugin-user-settings                  0.5.0
  @backstage/release-manifests                     0.0.6
  @backstage/test-utils                            1.2.1
  @backstage/theme                                 0.2.16
  @backstage/types                                 1.0.2
  @backstage/version-bridge                        1.0.3
Done in 0.87s.
```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by dhoizner** [comment]
I just went through the steps in the [react-router stable migration](https://github.com/backstage/backstage/blob/master/docs/tutorials/react-router-stable-migration.md) again and the issue is resolved, sorry!

Hopefully this helps anyone else that runs into this in the future.

Closing!

**Comment by painotpi** [comment]
> I just went through the steps in the [react-router stable migration](https://github.com/backstage/backstage/blob/master/docs/tutorials/react-router-stable-migration.md) again and the issue is resolved, sorry! I must have missed a step the first time around.
> 
> Hopefully this helps anyone else that runs into this in the future.
> 
> Closing!

In the following section of the doc,

>  Step 1 - Upgrade to Backstage 1.6
The first Backstage release to support react-router v6 is 1.6. You should upgrade to this version first before you start migrating. If you are an early bird and want to try out migration before that release, it is also shipped in 1.6.0-next.1.

I assume this means Backstage `v1.16` and `v1.16.0-next.1`?
